data_structures/binary_search_tree_OOP_approach.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * \file
+ * \brief A simple tree implementation using structured nodes
+ * \Object Oriented approach
+ */
+
+#include<bits/stdc++.h> 
```

Comment:
Don't use this library as it slows down the compilation process and is quite Linux-specified.